tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -35,45 +35,36 @@ void noc_async_write_multicast_one_packet_no_path_reserve(
 }
 
 FORCE_INLINE
-void multicore_cb_wait_front(bool db_buf_switch, int32_t num_pages) {
+void multicore_cb_wait_front(db_cb_config_t* db_cb_config, int32_t num_pages) {
     DEBUG_STATUS('C', 'R', 'B', 'W');
 
-    uint32_t pages_acked = *reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_ack_addr(db_buf_switch));
-    volatile tt_l1_ptr uint32_t* pages_received_ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_recv_addr(db_buf_switch));
-
     uint16_t pages_received;
     do {
-        pages_received = uint16_t(*pages_received_ptr) - pages_acked;
+        pages_received = uint16_t(db_cb_config->recv - db_cb_config->ack);
     } while (pages_received < num_pages);
     DEBUG_STATUS('C', 'R', 'B', 'D');
 }
 
 void multicore_cb_pop_front(
+    db_cb_config_t* db_cb_config,
+    const db_cb_config_t* remote_db_cb_config,
     uint64_t producer_noc_encoding,
-    bool db_buf_switch,
-    uint32_t fifo_limit,
-    uint32_t fifo_size,
-    uint32_t num_pages,
+    uint32_t consumer_fifo_limit,
+    uint32_t num_to_write,
     uint32_t page_size) {
-    volatile tt_l1_ptr uint32_t* CQ_CONSUMER_CB_ACK_PTR = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_ack_addr(db_buf_switch));
-    volatile tt_l1_ptr uint32_t* CQ_CONSUMER_CB_READ_PTR =
-        reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_rd_ptr_addr(db_buf_switch));
-
-    *CQ_CONSUMER_CB_ACK_PTR += num_pages;
-    *CQ_CONSUMER_CB_READ_PTR += (page_size * num_pages) >> 4;
+    db_cb_config->ack += num_to_write;
+    db_cb_config->rd_ptr += page_size * num_to_write;
```

Comment:
why was this changed? page_size is always in units of bytes, not 16B.

Reply:
Hmm I was just refactoring here, but the page size was 16B sized from before (this should be on main):
```
- reinterpret_cast<volatile tt_l1_ptr uint32_t*>(page_size_addr)[0] = page_size >> 4;
```